datayoga-py/datayoga/blocks/std/read/block.py

```diff
@@ -0,0 +1,42 @@
+import ast
+import logging
+import select
+import sys
+import uuid
+from typing import Any, Dict, Generator, List, Optional
+
+import aioconsole
+from datayoga.context import Context
+from datayoga.producer import Message
+from datayoga.producer import Producer as DyProducer
+
+logger = logging.getLogger("dy")
+
+
+class Block(DyProducer):
+
+    def init(self, context: Optional[Context] = None):
+        logger.debug(f"Initializing {self.get_block_name()}")
+
+    async def produce(self) -> Generator[Message, None, None]:
+        def get_records(data: str) -> List[Dict[str, Any]]:
```

Comment:
does this have to be nested within the method vs a class method?

Reply:
no, changed.